### DISCUSSION TITLE
Welcome to my Nix config

### DISCUSSION THREAD

**Discussion by dustinlyons** [dustinlyons/nixos-config#11] [Announcements]
## 👋 Welcome!
  I'm using Discussions as a place to connect with other folks using this code. The hope is that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Welcome others and are open-minded.

Thanks.

**Comment by StarrWulfe** [comment]
This is beautiful. 

I've been dabbling with Nix for almost 5 years both the package manager and the OS. I borked my Mac and wound up having to reinstall a fresh copy of Ventura just 2 days ago-- after having a play at trying to run NixOS on bare metal and seeing all the hardware limitations for this MacbookPro 14,2 model. I have an older 8,1 model that runs PopOS flawlessly and I was thinking "I could shift my efforts over to that machine and check it out, but I sure would like to investigate automating Homebrew since I have a fresh install again." which got me thinking *"There's gotta be a way someone used Nix to automate Brew* 🤔" And it led me here!

With your toolkit here, I think I can do both! 👏 

If I can get this working with both my Macbook-MacOS and MacBook-NixOS, then there shouldn't be any problems adding a third target PC/VM/Anything, right? How/Where to do that?

**Reply by dustinlyons** [reply]
Hey, thanks! You got it. You can add new targets in the flake.nix file, for either darwinConfigurations (if you're configuring Nix on Mac), or nixosConfigurations (if you're configuring NixOS).

For example, I want to add my home lab server; I'll be adding that to nixosConfigurations at some point. Then I just reference that target when I want to build it:

```
sudo nixos-rebuild switch --flake .#homeLab
```

**Comment by StarrWulfe** [comment]
OK.
I also had to wrap my mind around mas -- I kept getting 1Password and
Wireguard trying to download from the AppStore 😅  But I found a new way to
automate App Store downloads so thanks for that!

The big problem I'm having now is there's no way to do a nix-shell -p
*some-app* without me manually renaming the old nix.conf again then putting
it back out the way; is this normal behavior or is something wrong? I
should also note my Mac is an Intel on the amd64_darwin architecture so I
had to go through and change that everywhere too. Maybe I missed/broke
something?

**Reply by dustinlyons** [reply]
I generally try not to use any of the dash commands; those are before flakes. After flakes 'nix shell', 'nix develop', etc became available.

`nix shell nixpkgs#mas` should drop you in a shell with mas.

**Reply by StarrWulfe** [reply]
Got it.. It's a whole new world for me with flakes-- just as I got comfy with the legacy stuff here! 

_heads off to find a nix flakes cheatsheet_

**Comment by StarrWulfe** [comment]
OK, next question-- how do I burn it down and rebuild the whole thing if I screw up?

(I've got random permission errors happening in `/nix/store` and for some reason `bundle` is throwing gem related errors. I'd just like to tear it all down and restart again if possible...)

**Reply by dustinlyons** [reply]
The beauty of Nix is that you're not really doing anything destructive *with Nix* until you actually run the `darwin-rebuild` command, as before that, you're just building packages on the side and then doing a symlink switcharoo when the `darwin-rebuild` runs. 

I say that, however, `nix-homebrew` and the `homebrew` home manager module can be destructive, as it's just imperative homebrew commands under the hood. It's possible something happened there.

Here are uninstall steps for the Nix installer I use: https://zero-to-nix.com/start/uninstall

But that won't fix your homebrew situation, which may be what the `bundle` errors are about.

**Comment by StarrWulfe** [comment]
Alrighty, starting up again... and this time I get a cert error. I followed directions to edit the file that nix uses to know where CAcerts are, but it seems to have no effect. And it seems it wants to build everything from source now...? 

<img width="1153" alt="Screenshot 2023-10-19 at 6 29 41 PM" src="https://github.com/dustinlyons/nixos-config/assets/445009/dafa7955-4a47-42b8-b5ec-3b14af193a08">

Not sure how to recover from this one, except try, try again....

**Comment by dustinlyons** [comment]
Hey @StarrWulfe , I found this relevant thread which may solve your problem: https://discourse.nixos.org/t/ssl-ca-cert-error-on-macos/31171/3
